mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessage.java

```diff
@@ -234,20 +253,20 @@ public final boolean equals(Object o) {
                 && Objects.equals(this.messageId, that.messageId)
                 && Objects.equals(this.originMailboxes, that.originMailboxes)
                 && Objects.equals(this.owner, that.owner)
-                && Objects.equals(this.content, that.content)
                 && Objects.equals(this.deliveryDate, that.deliveryDate)
                 && Objects.equals(this.deletionDate, that.deletionDate)
                 && Objects.equals(this.sender, that.sender)
                 && Objects.equals(this.recipients, that.recipients)
-                && Objects.equals(this.subject, that.subject);
+                && Objects.equals(this.subject, that.subject)
+                && Objects.equals(Optional.ofNullable(this.content).map(DeletedMessage::hashContent), Optional.ofNullable(that.content).map(DeletedMessage::hashContent));
```

Comment:
it's worse than comparing both streams because you read them both fully and hash them.
My suggestion was to hash them only once.

Reply:
I understood, but `equals` is called only in test. Why would we need to compute the hash for in production?

Reply:
Then you can remove the content equals from this method & handle it in tests.
In a good immutable world, equals on messageId should be enough.

Reply:
See https://github.com/linagora/james-project/pull/2207 which fixes this raging debate